### PR TITLE
SONAR-9670 Introduce sonar.branch.longLivedBranches.regex

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: sonar-core/src/main/java/org/sonar/core/config/CorePropertyDefinitions.java [hunk 1]
```diff
@@ -41,6 +41,7 @@
   public static final String LEAK_PERIOD_MODE_PREVIOUS_VERSION = "previous_version";
 
   private static final String DEFAULT_LEAK_PERIOD = LEAK_PERIOD_MODE_PREVIOUS_VERSION;
+  private static final String DEFAULT_LONG_LIVED_BRANCHES_REGEX = "feature/.*";
```
Comment (janos-ss) [inline sonar-core/src/main/java/org/sonar/core/config/CorePropertyDefinitions.java]: I think feature branches are typically short-lived, so this is not a good default pattern for long-lived branches. I think that for example `(branch|release)-.*` would be better. We could look at some open-source projects that manage multiple branches to support multiple recent releases and see if there's a common pattern.

File: sonar-core/src/main/java/org/sonar/core/config/CorePropertyDefinitions.java [hunk 2]
```diff
@@ -235,6 +236,17 @@ private CorePropertyDefinitions() {
         .subCategory(CoreProperties.SUBCATEGORY_DUPLICATIONS_EXCLUSIONS)
         .multiValues(true)
         .build(),
+        
+        // BRANCHES
+        PropertyDefinition.builder(CoreProperties.LONG_LIVED_BRANCHES_REGEX)
+        .defaultValue("")
+        .name("Detection of long lived branches")
+        .description("Regular expression used to detect wheter a branch is a long living branch (as opposed to short living branch), based on its name.")
```
Comment (janos-ss) [inline sonar-core/src/main/java/org/sonar/core/config/CorePropertyDefinitions.java]: `s/wheter/whether/`

File: sonar-core/src/main/java/org/sonar/core/config/CorePropertyDefinitions.java (hunk 1 above)
Comment (dbmeneses) [inline sonar-core/src/main/java/org/sonar/core/config/CorePropertyDefinitions.java]: good point

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [event](https://github.com/SonarSource/sonarqube/blob/f9752c68e3265fd891ddf2f4fd86bfa6e237756f/sonar-plugin-api/src/main/java/org/sonar/api/scan/event): Add a 'package-info.java' file to document the 'event' package [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [event](https://github.com/SonarSource/sonarqube/blob/3c11790e3269b70a35c5884c9accd2a221835c4a/sonar-plugin-api/src/main/java/org/sonar/api/scan/event): Add a 'package-info.java' file to document the 'event' package [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [event](https://github.com/SonarSource/sonarqube/blob/8ea8558139cb7ef046a5ea12c5a71d7316c88840/sonar-plugin-api/src/main/java/org/sonar/api/scan/event): Add a 'package-info.java' file to document the 'event' package [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request closed without merging**